torch/_dynamo/variables/builtin.py

```diff
@@ -1212,9 +1212,13 @@ def _grad_changed(old, new):
                             else:
                                 grapharg.example.grad = None
                         return VariableBuilder(tx, source)(grapharg.example.grad)
-                unimplemented("tensor grad")
+
+                return obj.dynamic_getattr(tx, name)
             else:
-                unimplemented("tensor grad")
+                example_value = obj.as_proxy().node.meta["example_value"]
+                if example_value.grad is not None:
+                    unimplemented("getattr on non-None grad - NYI")
```

Comment:
maybe `"getattr on non-None grad coming from tensor that does not have a source"`?

Reply:
Sure :)